gwctl/pkg/printer/backends.go

```diff
@@ -28,7 +34,96 @@ import (
 )
 
 type BackendsPrinter struct {
-	Out io.Writer
+	Out   io.Writer
+	Clock clock.Clock
+}
+
+func (bp *BackendsPrinter) Print(resourceModel *resourcediscovery.ResourceModel) {
+	tw := tabwriter.NewWriter(bp.Out, 0, 0, 2, ' ', 0)
+	row := []string{"NAMESPACE", "NAME", "TYPE", "REFERRED BY ROUTES", "AGE", "POLICIES"}
+	tw.Write([]byte(strings.Join(row, "\t") + "\n"))
+
+	backendNodes := make([]*resourcediscovery.BackendNode, 0, len(resourceModel.Backends))
+	for _, backendNode := range resourceModel.Backends {
+		backendNodes = append(backendNodes, backendNode)
+	}
+
+	sort.Slice(backendNodes, func(i, j int) bool {
+		if backendNodes[i].Backend.GetNamespace() != backendNodes[j].Backend.GetNamespace() {
+			return backendNodes[i].Backend.GetNamespace() < backendNodes[j].Backend.GetNamespace()
+		}
+		return backendNodes[i].Backend.GetName() < backendNodes[j].Backend.GetName()
+	})
+
+	httpRouteNodes := make([]*resourcediscovery.HTTPRouteNode, 0, len(resourceModel.HTTPRoutes))
+	for _, httpRouteNode := range resourceModel.HTTPRoutes {
+		httpRouteNodes = append(httpRouteNodes, httpRouteNode)
+	}
+
+	for _, backendNode := range backendNodes {
+		backend := backendNode.Backend
+
+		parentHttpRoutes := []string{}
+		remainderHttpRoutes := 0
+		associatedDistinctPolicies := map[string]any{}
+
+		// the nested loops could've been made more efficient (on textbook) by using map between httpRouteNodes and backendRefs
+		// and leveraging the O(1) lookups but that seemed to sacrifice the code readability/structure, so didn't choose to do so
```

Comment:
According to me, creating maps would be helpful even if we lose some readability.

Reply:
I personally don't think so for the following reasons:
- the difference in runtime would be negligible majorly,  almost unrecognisable the end-user's experience.
- For significantly large use-cases (which would be rare), the map would lead to a much larger memory footprint.

Thank you for your suggestion but I'd rather incline towards the better readability and maintainability of this subroutine than giving in to a potentially premature optimization.

Rest let's wait for the maintainers to hear their opinions.

Cheers!